openwisp_radius/saml/views.py

```diff
@@ -0,0 +1,74 @@
+from urllib.parse import parse_qs, urlparse
+
+import swapper
+from django.core.exceptions import ImproperlyConfigured, ObjectDoesNotExist
+from django.shortcuts import get_object_or_404
+from djangosaml2.views import (
+    AssertionConsumerServiceView as BaseAssertionConsumerServiceView,
+)
+from djangosaml2.views import (  # noqa
+    LoginView,
+    LogoutInitView,
+    LogoutView,
+    MetadataView,
+)
+from rest_framework.authtoken.models import Token
+
+from ..api.views import RadiusTokenMixin
+from ..utils import load_model
+from .utils import get_url_or_path
+
+Organization = swapper.load_model('openwisp_users', 'Organization')
+RadiusToken = load_model('RadiusToken')
+RegisteredUser = load_model('RegisteredUser')
+OrganizationUser = swapper.load_model('openwisp_users', 'OrganizationUser')
+
+
+class AssertionConsumerServiceView(RadiusTokenMixin, BaseAssertionConsumerServiceView):
+    def get_organization_from_relay_state(self):
+        try:
+            parsed_url = urlparse(self.request.POST.get('RelayState'), None)
+            org_slug = parse_qs(parsed_url.query)['org'][0]
+        except (KeyError, IndexError):
+            raise ImproperlyConfigured('Organization slug not provided')
+        else:
+            return get_object_or_404(Organization, slug=org_slug)
+
+    def post_login_hook(self, request, user, session_info):
+        """ If desired, a hook to add logic after a user has successfully logged in.
+        """
+        org = self.get_organization_from_relay_state()
+        is_member = user.is_member(org)
+        # add user to organization
+        if not is_member:
+            orgUser = OrganizationUser(organization=org, user=user)
+            orgUser.full_clean()
+            orgUser.save()
+        try:
+            user.registered_user
+        except ObjectDoesNotExist:
+            registered_user = RegisteredUser(
+                user=user, method='social_login', is_verified=False
```

Comment:
`social_login` > `saml`, you need to register this new method but only if saml is enabled.